binary/cardano-binary.cabal

```diff
@@ -60,7 +60,12 @@ library
                        -Wredundant-constraints
                        -Wpartial-fields
                        -Wcompat
-
+  if impl(ghc >=8.8)
+    ghc-options:       -Wno-missing-deriving-strategies
+  if impl(ghc >=8.10)
+    ghc-options:       -Wno-missing-safe-haskell-mode
+                       -Wno-prepositive-qualified-module
+                       -Wno-unused-packages
```

Comment:
Actually, are all these flags that disable warnings even needed (in the other cabal files as well)? I assume you have copied them from `cardano-prelude`, but there `-Weverything` is used. Here `-Wall` is used, so these extra warnings don't have to be disabled. (I didn't have to disable them in https://github.com/input-output-hk/ouroboros-network/pull/2542)

I prefer `-Wall` + a manual selection of flags over `-Weverything`, which forces you to disable some flags based on the GHC version.

Reply:
Hmm, certainly some of them were needed in some places. But it's true that after adding them to a couple of files, I pre-emptively added them in other places...

Reply:
BTW, I wouldn't disable `-Wno-unused-packages`, I'd just fix the warnings :slightly_smiling_face:

Reply:
I just tried removing all those flags myself and it compiled without warnings (except for those from the C code).

Reply:
> I prefer `-Wall` + a manual selection of flags over `-Weverything`, which forces you to disable some flags based on the GHC version.

In other words, no changes are needed to the cabal files at all.